### DISCUSSION TITLE
Rendering in Quarto

### DISCUSSION THREAD

**Discussion by friosavila** [hugetim/nbstata#16] [General]
Hi Tim, 
Just a quick question (which i have also sent to quarto dev). 
Not sure if you kept exploring quarto, but i encounter an odd problem when rendering documents straight from Stata.

consider this

````
---
title: "Untitled"
format: 
    html: default
    ipynb: default
    pdf: default
jupyter: nbstata
---

First setup

```{stata}
clear
ssc instal jwdid
set linesize 250
set seed     111
set sortseed 111
set obs 100  // <- 100 units
gen id = _n
gen ai = rchi2(2)
// determines When would units receive treatment
gen     g = runiformint(2,10)
replace g = 0 if g>9   // never treated       
expand 10   // <-T=10
bysort id:gen t=_n 
gen event = max(0,t-g)
gen aux = runiform()*2
bysort t:gen at = aux[1] // Determines Time fixed effect
gen te = (1-g/10)+(1-event/10)  
// Treatment effect but vanishes with time
gen eit= rnormal()
gen trt  = (t>=g)*(g>0)
gen teff = te * trt 
gen y = ai + at + te * trt + eit
```

then 2 regressions

```{stata}
qui: jwdid y, ivar(i) tvar(t) gvar(g)
estat simple
```

but also

```{stata}
qui: jwdid y, ivar(i) tvar(t) gvar(g) never
estat simple
```
````

So when I render this into HTML i get two outcomes from the regressions:

a good one

```
------------------------------------------------------------------------------
             |            Delta-method
             | Coefficient  std. err.      z    P>|z|     [95% conf. interval]
-------------+----------------------------------------------------------------
         _at |
   (2 vs 1)  |   1.543749   .1567784     9.85   0.000     1.236469    1.851029
------------------------------------------------------------------------------
```

but also one misaligned

```
------------------------------------------------------------------------------
             |            Delta-method
             | Coefficient  std. err.      z    P>|z|     [95% conf. interval]
-------------+----------------------------------------------------------------
         _at |
   (2 vs 1)  |
   1.458058   .1494798     9.75   0.000     1.165083    1.751033
------------------------------------------------------------------------------
```

Do you have any idea why would this happen?

**Comment by hugetim** [comment]
[Note: I made a small edit to your comment to fix the formatting: adding an extra tick to the enclosing ticks: ````]

My short answer is: No, I don't know.

I have been using Quarto, though. Just yesterday I spent a while trying to get some html rendering how I wanted it. My frustration was that it interprets "asis" output as Markdown rather than raw HTML as I would like.

**Reply by friosavila** [reply]
Hi Tim
The Developer in Quarto seems to believe the problem may be related to the nbclient engine. 
here what they say:
"I suspect the issue arises from the fact that we're using nbclient as the python library to execute the jupyter engines, and there's some difference in the way nbclient sets up the engines compared to jupyter."

**Reply by hugetim** [reply]
For what it's worth, I finally figured out my html issue. This (that is, the surrounding `disp` commands) works, in a Stata code cell, to render the esttab html correctly (including model numbers and the stars footnote):

```{stata}
disp "```{=html}"
esttab m1 m2 m3, mtitles keep(treat_effect yq) se align(center) width(100%) html
disp "```"
```

**Reply by hugetim** [reply]
Actually, it doesn't always work. I don't know if the problem is with `esttab` or something in the Quarto stack, but within the surrounding `disp` commands, `esttab, nostar` no longer renders correctly.

**Comment by friosavila** [comment]
That is good to know!
I should try that instead of Markdown tables.

Im still trying to see how to solve the other problem I mentioned. when
pritint output straight out of stata. for some reason things get messed up
in quatro, or quatro produced jupyter notebooks, but not if I do it in
jupyter notebooks directly.
 anyways, I ll try your trick!

**Comment by friosavila** [comment]
something you may want to try
<iframe  width=100%   height =2000px src="aux.html"></iframe>
with the src being a file created by esttab.
The only thing i do not know is how to request Height to be automatically
sized.

**Comment by friosavila** [comment]
To add to the previous point regarding code output.

I figure out a way around it.

Display it with two codes:
1st. Display the code of interest
```stata
```

2nd. display the output as is
```{stata}
*| echo: off
display "```"
All code to be shown.
display "```"
```

Makes it for longer coding, but its at least an improvement.
F
